opensrp-chw-core/src/main/java/org/smartregister/chw/core/task/UpdateBirthNotificationLastAsyncTask.java

```diff
@@ -0,0 +1,153 @@
+import org.jeasy.rules.api.Rules;
+import org.smartregister.chw.core.R;
+import org.smartregister.chw.core.application.CoreChwApplication;
+import org.smartregister.chw.core.dao.VisitDao;
+import org.smartregister.chw.core.domain.VisitSummary;
+import org.smartregister.chw.core.holders.RegisterViewHolder;
+import org.smartregister.chw.core.model.ChildVisit;
+import org.smartregister.chw.core.utils.ChwDBConstants;
+import org.smartregister.chw.core.utils.CoreChildUtils;
+import org.smartregister.chw.core.utils.CoreConstants;
+import org.smartregister.commonregistry.CommonPersonObject;
+import org.smartregister.commonregistry.CommonRepository;
+import org.smartregister.family.util.DBConstants;
+import org.smartregister.family.util.Utils;
+
+import java.text.SimpleDateFormat;
+import java.util.Locale;
+import java.util.Map;
+
+import timber.log.Timber;
+
+import static org.smartregister.chw.core.utils.Utils.getDuration;
+
+public class UpdateBirthNotificationLastAsyncTask extends AsyncTask<Void, Void, Void> {
+    public final Context context;
+    private final CommonRepository commonRepository;
+    public final RegisterViewHolder viewHolder;
+    public final String baseEntityId;
+    private final Rules rules;
+    public CommonPersonObject commonPersonObject;
+    public ChildVisit childVisit;
+    public View.OnClickListener onClickListener;
+    private SimpleDateFormat ISO8601DATEFORMAT = new SimpleDateFormat("yyyy-MM-dd", Locale.ENGLISH);
+
+    public UpdateBirthNotificationLastAsyncTask(Context context, CommonRepository commonRepository, RegisterViewHolder viewHolder, String baseEntityId, View.OnClickListener onClickListener) {
+        this.context = context;
+        this.commonRepository = commonRepository;
+        this.viewHolder = viewHolder;
+        this.baseEntityId = baseEntityId;
+        this.onClickListener = onClickListener;
+        this.rules = CoreChwApplication.getInstance().getRulesEngineHelper().rules(CoreConstants.RULE_FILE.HOME_VISIT);
+    }
+
+    @Override
+    public Void doInBackground(Void... params) {
+        if (commonRepository != null) {
+            commonPersonObject = commonRepository.findByBaseEntityId(baseEntityId);
+
+            Map<String, VisitSummary> map = VisitDao.getVisitSummary(baseEntityId);
+            if (map != null) {
+                VisitSummary notDoneSummary = map.get(CoreConstants.EventType.CHILD_VISIT_NOT_DONE);
+                VisitSummary lastVisitSummary = map.get(CoreConstants.EventType.CHILD_HOME_VISIT);
+
+                long lastVisit = 0;
+                long visitNot = 0;
+                long dateCreated = 0;
+                try {
+                    String createVal = Utils.getValue(commonPersonObject.getColumnmaps(), ChwDBConstants.DATE_CREATED, false);
+                    if (StringUtils.isNotBlank(createVal))
+                        dateCreated = ISO8601DATEFORMAT.parse(createVal).getTime();
+
+                } catch (Exception e) {
```

Comment:
Any reason why you can capture a less generic exception?

Reply:
Yes, it can produce multiple exceptions like ParseException or NullPointerException.